src/NCronJob/Scheduler/JobWorker.cs

```diff
@@ -84,7 +84,7 @@ private async Task DispatchJobForProcessing(
 
         try
         {
-            var cts = jobQueueManager.GetOrAddCancellationTokenSource(queueName);
+            var cts = jobQueueManager.GetCancellationTokenSource(queueName);
             using var linkedCts = CancellationTokenSource.CreateLinkedTokenSource(cts.Token, cancellationToken);
             var linkedToken = linkedCts.Token;
 
```

Comment:
After the `WaitForNextExecution` - shouldn't we also check if `jobQueueManager` is already disposed? Not related to your PR - but it seems fishy.

Reply:
I agree. However this might require some deeper investigation. Logged as #186